src/SFA.DAS.EmployerFinance/Queries/GetTransferTransactionDetails/GetTransferTransactionDetailsQueryHandler.cs

```diff
@@ -50,9 +51,18 @@ public async Task<GetTransferTransactionDetailsResponse> Handle(GetTransferTrans
                 ApprenticeCount = (uint)ct.DistinctBy(t => t.CommitmentId).Count()
             }).ToArray();
 
-            var periodEnd = _dbContext.PeriodEnds.Single(p => p.PeriodEndId.Equals(firstTransfer.PeriodEnd));
+            //NOTE: We should only get one tranfer transaction per sender per period end
+            // as this is how transfers are grouped together when creating transfer transactions
+            var tranferTransaction = _dbContext.Transactions.Single(t =>
```

Comment:
Does this query know that it will be able find a single row? The comment states that having more than one is an error but might there be zero matching rows legitimately?